File: public/usage-examples/animations/animation_current_vector-1-example.py
from splashkit import *

# Load explosion animation script and create animation
script = load_animation_script("ExplosionScript", "explosion.txt")
anim = create_animation(script, "Explode")

# Get vector for current frame
vec = animation_current_vector(anim)
write_line(f"Current vector: x={vec.x} y={vec.y}")

free_animation(anim)
free_animation_script(script)

File: public/usage-examples/animations/animation_current_vector-1-example.cpp
#include "splashkit.h"

int main()
{
    // Load the animation script containing vector annotations (explosion.txt)
    animation_script script = load_animation_script("ExplosionScript", "explosion.txt");
    animation anim = create_animation(script, "Explode");

    // In a running animation you can query the current frame's vector
    vector_2d v = animation_current_vector(anim);
    write_line("Current vector: x=%f y=%f", v.x, v.y);

    free_animation(anim);
    free_animation_script(script);
    return 0;
}
